### PR TITLE
Adds menu entry swaps for Buy 1/5/10/50 and Sell 1/5/10/50.

### PR DESCRIPTION
**Pull request by geheur**
Pull request for issue #11427.

### CODE REVIEW

**Comment by dekvall** [comment]
This seems dangerous, especially the default sell setting which some users will undoubtedly forget to reset their setting after selling something in bulk and then end up selling their gear to the general store. Some specific items have had their buy prices [artificially inflated](https://www.reddit.com/r/2007scape/comments/31gus9/dfs_costs_2b_in_general_store/) and might be hard to get back once sold.

I'm thinking a better solution would be to have an option to remember the last used Sell option and reset it every time you open a shop interface. I can see how the use of a configurable buy option is beneficial to ironmen who needs to run back and forth from shops a lot though.

I don't know of many situations where hopping and selling items are done except for the karamja store, and even then it would only be one extra click per hop in comparison to a lost dfs.

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/BuyMode.java [hunk 1]
```diff
@@ -0,0 +1,47 @@
...
+	BUY_10("Buy 10"),
+	BUY_50("Buy 50");
+
+	private final String name;
+
+	@Override
+	public String toString()
+	{
+		return name;
+	}
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/BuyMode.java:46]: All of this can be removed. [Enum names are converted to Title-Cased Entries in the resulting combobox.](https://github.com/runelite/runelite/blob/8eb57c5a92ee05df47dbdc9d530f3c2da77f0bd4/runelite-client/src/main/java/net/runelite/client/ui/components/ComboBoxListRenderer.java#L62-L65)
```suggestion
```

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/BuyMode.java [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+ * SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+package net.runelite.client.plugins.menuentryswapper;
+
+import lombok.Getter;
+import lombok.RequiredArgsConstructor;
+
+@Getter
+@RequiredArgsConstructor
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/BuyMode.java:31]: Similarly, these are not needed.
```suggestion
```

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/SellMode.java [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+	SELL_10("Sell 10"),
+	SELL_50("Sell 50");
+
+	private final String name;
+
+	@Override
+	public String toString()
+	{
+		return name;
+	}
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/SellMode.java:46]: See my comments on `BuyMode`.

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java [hunk 4]
```diff
@@ -773,6 +773,46 @@ else if (option.equals("collect to inventory") || option.startsWith("collect-not
...
+					case BUY_5:
+						swap("buy 5", option, target, index);
+						break;
+					case BUY_10:
+						swap("buy 10", option, target, index);
+						break;
+					case BUY_50:
+						swap("buy 50", option, target, index);
+						break;
+				}
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java:794]: Instead of creating a switch case for each enum value, the menu options can be defined in the enum and set much more concisely here.
```suggestion
				swap(config.shopBuy().getOption(), option, target, index);
```

**Comment by geheur** [comment]
I didn't consider how dangerous this could be if you forgot you left it enabled.

I think the safest way to do it would be to set a buy/sell quantity on a per-item basis and have it activated through some kind of shift-rightclick menu. If you're buying or selling so much of something that you'd go through the effort of setting a sell/buy amount for an item, it's unlikely that buying or selling 1/5/10/50 of that item on accident in the future would cripple you financially.

I don't know if individual buy/sell quantities is considered "overpowered" by jagex but if it is then you could enable the swap for individual items and have the same quantity for each item where this is enabled, with the quantity being set in the plugin config.

Resetting the sell setting every time a shop interface would address a few problematic situations, but if you forget you have this enabled and sell an item, and then try to "value" another item, you'd still sell it accidentally.

**Comment by Nightfirecat** [comment]
I think it's fine to leave in. In my experience, players are buying from shops much more often than selling to them, and if they're selling to shops, it's usually only very specific items to specific shops--usually to those that offer high-alch prices.

**Comment by geheur** [comment]
@Nightfirecat what exactly are you referring to as being safe to leave in? The way it currently is on my branch (option in config to set buy/sell 1/5/10/50 for all items), dekvall's suggestion (reset this setting every time a shop interface is opened), or my suggestion (remember sell/buy quantity on a per-item basis)?

**Comment by Nightfirecat** [comment]
I mean it should be fine to leave as-is currently PR'd.

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/BuyMode.java (hunk 2 above)
Comment (geheur) [reply runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/BuyMode.java:31]: This should not be removed if the option needs to be accessible to make code in MenuEntrySwapperPlugin more concise.

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java (hunk 4 above)
Comment (geheur) [reply runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java]: Added this for other similar menu entry swaps.

**Comment by geheur** [comment]
I think the risk of accidentally buying or selling something you don't want it still really high - I wouldn't trust myself to use this as it is now even knowing the dangers.

I plan on making this setting unique to each item, but feel free to accept this as it is now if y'all prefer that.

**Comment by geheur** [comment]
Alright, I'm happy with the way this works now. The swap is now specific to the item and whether it's a Buy or a Sell.

**Comment by Adam-** [comment]
I don't see anything wrong with having a sell-bulk option and a buy-bulk option and just defaulting them to 1?

**Comment by geheur** [comment]
Reverted all of the item-specific stuff, and changed the swap to only occur when shift is held, to prevent accidental buying or selling.

**Comment by Nightfirecat** [comment]
Why is this PR `+557 −462`? Please revert everything except the feature you're adding.

**Comment by geheur** [comment]
Alright, took out everything not strictly related to this change.

The reason this PR looked big earlier was because I:

Took carriage returns out of a file (MenuEntrySwapperConfig.java), since either intellij or git was automatically removing them and that was causing me some problems, and it seemed like most other files in the project had unix-style line endings.

Changed some other unrelated menu entry swaps to be more concise in the same way you suggested for the buy/sell swap in this PR (getting the option string from the Enum instead of a switch statement.

Sorry if doing any of that was a faux pas, or if it should have been done differently.

File: runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java [hunk 5]
```diff
@@ -884,7 +896,7 @@ private int findIndex(MenuEntry[] entries, int limit, String option, String targ
 	{
 		if (strict)
 		{
-			List<Integer> indexes = optionIndexes.get(option);
+			List<Integer> indexes = optionIndexes.get(option.toLowerCase());
```
Comment (Nightfirecat) [inline runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java:899]: Why is this changed?

Comment (geheur) [reply runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java:899]: I changed this so that I could have the option strings in BuyMode and Sellmode have the same capitalization as they do in-game, rather than already being in lower case. Should I change BuyMode and Sellmode to just have the option be lowercase to start with?

Comment (Nightfirecat) [reply runelite-client/src/main/java/net/runelite/client/plugins/menuentryswapper/MenuEntrySwapperPlugin.java:899]: That would be best. :+1:
